src/main/java/org/tinyradius/packet/RadiusPacket.java

```diff
@@ -1007,7 +1018,7 @@ protected static RadiusPacket decodePacket(Dictionary dictionary, InputStream in
 			throw new RadiusException("bad packet: attribute length mismatch");
 
 		// create RadiusPacket object; set properties
-		RadiusPacket rp = createRadiusPacket(type);
+		RadiusPacket rp = createRadiusPacket((forceType == UNDEFINED) ? type : forceType);
```

Comment:
This seems a bit forced.  Why don't you remove the "forceType" parameter from decodePacket() and let the caller of it (RadiusServer.java#375) set the type on the returned value?  This will eliminate a bunch of code and doesn't leave us with this confusing piece of condition.